crates/ruff_linter/src/rules/flake8_quotes/rules/avoidable_escaped_quote.rs

```diff
@@ -231,7 +289,118 @@ pub(crate) fn avoidable_escaped_quote(
+                let fixed_contents = format!(
+                    "{prefix}{quote}{value}{quote}",
+                    prefix = kind.as_str(),
+                    quote = leading.as_char(),
+                    value = unescape_string(string_contents, leading.opposite().as_char())
+                );
+                diagnostic.set_fix(Fix::safe_edit(Edit::range_replacement(
+                    fixed_contents,
+                    tok_range,
+                )));
+                diagnostics.push(diagnostic);
+            }
+            Tok::FStringStart => {
+                let text = locator.slice(tok_range);
+                // Check for escaped quote only if we're using the preferred quotation
+                // style and it isn't a triple-quoted f-string.
+                let check_for_escaped_quote = !is_triple_quote(text);
+                let quote_style = if text.contains(Quote::Single.as_char()) {
+                    Quote::Single
+                } else {
+                    Quote::Double
+                };
+                fstrings.push(FStringContext::new(
+                    check_for_escaped_quote,
+                    tok_range,
+                    quote_style,
+                ));
+            }
+            Tok::FStringMiddle {
+                value: string_contents,
+                is_raw,
+            } if !is_raw => {
+                let Some(context) = fstrings.last_mut() else {
+                    continue;
+                };
+                if !context.check_for_escaped_quote {
+                    continue;
+                }
+                if string_contents.contains(context.quote_style.opposite().as_escaped()) {
+                    context.push_fstring_middle_range(tok_range);
+                }
+            }
+            Tok::FStringEnd => {
+                let Some(context) = fstrings.pop() else {
+                    continue;
+                };
+                if context.middle_ranges_with_escapes.is_empty() {
+                    // There are no `FStringMiddle` tokens containing any escaped
+                    // quotes.
+                    continue;
+                }
+                let mut diagnostic = Diagnostic::new(
+                    UnnecessaryEscapedQuote,
+                    TextRange::new(context.start_range.start(), tok_range.end()),
+                );
+                let mut fstring_middle_edits =
+                    context.middle_ranges_with_escapes.iter().map(|&range| {
+                        Edit::range_replacement(
+                            unescape_string(
+                                locator.slice(range),
+                                context.quote_style.opposite().as_char(),
+                            ),
+                            range,
+                        )
+                    });
+                diagnostic.set_fix(Fix::safe_edits(
+                    fstring_middle_edits.next().unwrap(),
```

Comment:
Is there a nicer way to pass this list of edits without having to manually consume the first item and pass it separately?

Not sure if Rust has something like Python's `*args` that could be used in `Fix::safe_exits(...)`...

Reply:
I did it slightly more manually, but in a way that combines it with the validation condition above (to ensure that we _must_ have at least one edit, and avoid runtime panics).